cocotb/scheduler.py

```diff
@@ -752,20 +757,20 @@ def schedule(self, coroutine, trigger=None):
         # chaining
         if coro_completed:
             self.unschedule(coroutine)
-            return
 
         # Don't handle the result if we're shutting down
         if self._terminate:
             return
 
-        try:
-            result = self._trigger_from_any(result)
-        except TypeError as exc:
-            # restart this coroutine with an exception object telling it that
-            # it wasn't allowed to yield that
-            result = NullTrigger(outcome=outcomes.Error(exc))
+        if not coro_completed:
```

Comment:
@dmiller-jumptrading, I don't really understand the motivation for this change. Happy to put it in, but I can't think of a case where it makes a difference.

Reply:
Worked it out myself at least, and your tests fail without it.